helm-charts/support/values.yaml

```diff
@@ -1,30 +1,83 @@
-prometheusIngressAuthSecret:
-  enabled: false
-  username: ""
-  password: ""
-
-redirects:
-  rules: []
+# Configuration of dependency charts
+# -------------------------------------------------------------------------------
+#
 
+# cluster-autoscaler is responsible for understanding if k8s nodes needs to be
+# added or removed. A pending pod is a sign that another node should be added,
+# and an underused node is a sign that a node should be removed.
+#
+# We always need a cluster-autoscaler, but only deploy it ourselves if its not
+# provided as part of the k8s cluster.
+#
+# values ref: https://github.com/kubernetes/autoscaler/blob/master/charts/cluster-autoscaler/values.yaml
+#
 cluster-autoscaler:
   enabled: false
+
+# ingress-nginx is responsible for proxying traffic based on k8s Ingress
+# resources defined in the k8s cluster.
+#
+# Typically, all inbound traffic arrives to some ingress-nginx controller pod
+# via a k8s Service that has a public IP, and is thereafter proxied to a k8s
+# Service (a Pod selected by a k8s Service) that doesn't have a public IP.
+#
+# values ref: https://github.com/kubernetes/ingress-nginx/blob/main/charts/ingress-nginx/values.yaml
+#
 ingress-nginx:
   controller:
     podLabels:
-      # nginx-ingress controllers need to communicate with the proxy pod (and only the proxy pod)
-      # in clusters with networkpolicy enforcement enabled. Adding this label on the controller pod
-      # allows that. https://z2jh.jupyter.org/en/stable/administrator/security.html#introduction-to-the-chart-s-four-network-policies
-      # has more information.
+      # nginx-ingress controllers need to be allowed proxy traffic onwards to
+      # JupyterHub's proxy pod (and only the proxy pod) in clusters with
+      # NetworkPolicy enforcement enabled. Adding this label on the controller
+      # pod allows that.
+      #
+      # ref: https://z2jh.jupyter.org/en/stable/administrator/security.html#introduction-to-the-chart-s-four-network-policies
+      #
       hub.jupyter.org/network-access-proxy-http: "true"
+
+# prometheus is responsible for collecting metrics and informing grafana about
+# the metrics on request. It comes with several dependency charts, where we
+# opt-out of a few.
+#
+# values ref: https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/values.yaml
+#
```

Comment:
I added a comments that I think are helpful to have in a values.yaml file. I find it acceptable if this summarizing descriptions are duplicated as they are supposed to be providing just a very high level overview.